modules/images/dominant-color/load.php

```diff
@@ -303,15 +274,25 @@ function dominant_color_has_transparency( $attachment_id ) {
 	if ( is_wp_error( $editor ) ) {
 		return $editor;
 	}
+
 	if ( ! method_exists( $editor, 'has_transparency' ) ) {
 		return new WP_Error( 'unable_to_find_method', __( 'Unable to find has_transparency method', 'performance-lab' ) );
 	}
 	$has_transparency = $editor->has_transparency();
-	if ( is_wp_error( $has_transparency ) ) {
-		return $has_transparency;
+	if ( ! is_wp_error( $has_transparency ) ) {
+		$dominant_color_data['has_transparency'] = $has_transparency;
+	}
+
+	if ( ! method_exists( $editor, 'get_dominant_color' ) ) {
+		return new WP_Error( 'unable_to_find_method', __( 'Unable to find get_dominant_color method', 'performance-lab' ) );
+	}
+
+	$dominant_color = $editor->get_dominant_color();
+	if ( ! is_wp_error( $dominant_color ) ) {
+		$dominant_color_data['dominant_color'] = $dominant_color;
 	}
```

Comment:
Same here, this should return the error instead.
```suggestion
	if ( is_wp_error( $dominant_color ) ) {
		return $dominant_color;
	}
	$dominant_color_data['dominant_color'] = $dominant_color;
```

Reply:
changed